app/src/main/java/tech/pauly/findapet/discover/AnimalDetailsViewPagerAdapter.java

```diff
@@ -0,0 +1,93 @@
+package tech.pauly.findapet.discover;
+
+import android.content.Context;
+import android.databinding.DataBindingUtil;
+import android.databinding.ViewDataBinding;
+import android.support.v4.view.PagerAdapter;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import javax.inject.Inject;
+
+import tech.pauly.findapet.R;
+import tech.pauly.findapet.databinding.ViewAnimalDetailsDetailsBinding;
+import tech.pauly.findapet.databinding.ViewAnimalDetailsLocationBinding;
+import tech.pauly.findapet.dependencyinjection.ForApplication;
+
+public class AnimalDetailsViewPagerAdapter extends PagerAdapter {
+
+    private final Context context;
+    private AnimalDetailsViewModel viewModel;
+
+    @Inject
+    AnimalDetailsViewPagerAdapter(@ForApplication Context context) {
```

Comment:
Pretty happy with how this turned out with multiple types instead of just the one in the DiscoverAdapter. Kind of boilerplate-y, but overall a good approach I think.